src/collections/maple_tree/mod.rs

```diff
@@ -273,5 +742,285 @@ mod tests {
+        rn.value[1] = M::Some(5);
+
+        rn.pivot[2] = M::Some(6);
+        rn.value[2] = M::Some(23);
+
+        rn.pivot[3] = M::Some(7);
+        rn.value[3] = M::Some(5);
+
+        rn.pivot[4] = M::Some(17);
+        rn.value[4] = M::Some(2);
+
+        rn.pivot[5] = M::Some(19);
+        rn.pivot[6] = M::None;
+        
+        assert!(rn.move_pivots_left(3, 3) == true);
+        check_node_state([7, 17, 19, 0, 0, 0, 0],[ 5, 2, 0, 0, 0, 0, 0, 0], &rn);
+
+        assert!(rn.move_pivots_left(3,4) == false);
+
+        assert!(rn.move_pivots_left(10, 3) == false);
+
+        assert!(rn.move_pivots_left(12, 15) == false);
+
+        assert!(rn.move_pivots_left(0, 1) == false);
+
+        assert!(rn.move_pivots_left(1, 1) == true);
+        check_node_state([17, 19, 0, 0, 0, 0, 0],[ 2, 0, 0, 0, 0, 0, 0, 0], &rn);
+
+        assert!(rn.move_pivots_left(1, 1) == false);
+        check_node_state([17, 19, 0, 0, 0, 0, 0],[ 2, 0, 0, 0, 0, 0, 0, 0], &rn);
+    }
+
+    // since RangeNode::move_pivots_right
+    // is used in RangeNode::insert we will test this function by
+    // inserting the pivots and values manually
+    #[test]
+    fn test_range_node_move_pivots_right(){
+        
+        let mut rn: RangeLeaf<usize, usize> = RangeLeaf::new();
+        rn.pivot[0] = M::Some(6);
+         
+        assert!(rn.move_pivots_right(0, 2) == true);
+        for i in 0..R_CAPACITY{
+            if i == 2{
+                assert!(rn.pivot[2] == M::Some(6));
+            }
+            else{
+                assert!(rn.pivot[i] == M::None);
+            }
+        }
+
+        let testOne: [usize; R_CAPACITY] = [1, 2, 6, 7, 0, 17, 19];  
+
+        rn.pivot[0] = M::Some(1);
+        rn.pivot[1] = M::Some(2); 
+        rn.pivot[2] = M::Some(6);
+        rn.pivot[3] = M::Some(7);
+        rn.pivot[4] = M::Some(17);
+        rn.pivot[5] = M::Some(19);
+        rn.pivot[6] = M::None;
+        
+        assert!(rn.move_pivots_right(4, 1) == true);
+
+
+        for i in 0..R_CAPACITY{
+            if testOne[i] == 0{
+                assert!(rn.pivot[i] == M::None);
+            }
+            else{
+                println!("{}", i);
+                assert!(rn.pivot[i] == M::Some(testOne[i]));
+            }
+        }
+        
+        let testTwo: [usize; R_CAPACITY] = [1, 2, 6, 7, 10, 15, 20];
+
+
+        for i in 0..R_CAPACITY{
+            rn.pivot[i] = M::Some(testTwo[i]);
+        }
+
+        assert!(rn.move_pivots_right(4,1) == false);
+
+        let testThree: [usize; R_CAPACITY] = [1, 2, 6, 7, 10, 15, 0];
+
+        for i in 0.. R_CAPACITY-1{
+            if testThree[i] == 0{
+                rn.pivot[i] = M::None;
+            }
+            else{
+                rn.pivot[i] = M::Some(testThree[i]);
+            }            
+        }
+
+        assert!(rn.move_pivots_right(2, 3) == false);
+
+    }
+    
+    fn check_node_state(pivots: [usize; R_CAPACITY], values: [usize; CAPACITY], node: &RangeLeaf<usize, usize> ){
```

Comment:
This is a super cool idea for verification/testing. You could mark this function as #[cfg(test)] so it's not used in production.